src/main/java/com/socialmediaraiser/twitterbot/impl/personalAnalyzer/PersonalAnalyzerBot.java

```diff
@@ -92,25 +96,25 @@ else if(showFollowers && showFollowings && onyFollowBackUsers){
         }
     }
 
-    private UserInteractions getNbInterractions(String archiveFileName) throws IOException {
-        File file = new File(getClass().getClassLoader().getResource(archiveFileName).getFile());
+    private UserInteractions getNbInterractions(String archiveFileName) {
         UserInteractions userInteractions = new UserInteractions();
+        Map<String, TweetInteraction> receivedInteractions = this.getReceivedInteractions();
         // counts all retweets given to others
-        dataArchiveHelper.countGivenRetweets(userInteractions);
+        dataArchiveHelper.countRetweetsGiven(userInteractions);
         // counts all the unique replies given by the user to others
         dataArchiveHelper.countRepliesGiven(userInteractions);
-        // counts all the retweets of user tweets done by others
-        dataArchiveHelper.countRetweesReceived(userInteractions);
         // counts all replies given recently to others
         apiSearchHelper.countRecentRepliesGiven(userInteractions, dataArchiveHelper.filterTweetsByRetweet(false).get(0).getCreatedAt()); // @todo test 2nd arg
-        // counts all the replies received by others
-        apiSearchHelper.countRepliesReceived(userInteractions, true); // D-7 -> D0
-        apiSearchHelper.countRepliesReceived(userInteractions, false); // D-30 -> D-7
         apiSearchHelper.countGivenLikesOnStatuses(userInteractions);
         return userInteractions;
     }
 
-
+    private Map<String, TweetInteraction> getReceivedInteractions(){
+        return apiSearchHelper.countRepliesReceived(true)
+                              .merge(apiSearchHelper.countRepliesReceived(false),
+                                     TweetInteraction::merge)
+                              .merge(dataArchiveHelper.countRetweetsReceived());
```

Comment:
Isn't it a bit more readable like that?

Reply:
Elegant !!

Reply:
Looks not working `TweetInteraction is not a functional interface`. Gonna check why tonight if you don't know neither.

Reply:
Is there a reason that the 2nd `.merge` doesn't have the `TweetInteraction::merge` 2nd arg ?